app/models/poly_metadatum.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# A polymetadatum can be assigned to any metadatable record.
+class PolyMetadatum < ApplicationRecord
+  # Associations
+  belongs_to :metadatable, polymorphic: true, optional: false
+
+  # Validations
+  validates :key, presence: true # otherwise nil is a valid key
+  validates :value, presence: true
```

Comment:
Is it intentional that the `presence: true` is not enforced at the database level?

Reply:
No, an oversight, have set to not null, good spot